### PR TITLE
[NEW] Disable registration

### PR DESCRIPTION
**Pull request by sowenjub**
@RocketChat/ios

This PR adds a new RC_DISABLE_REGISTRATION option to the AppManager, which can be useful for private servers where self-registration isn't allowed.
It simply hides all views related to the creation of a new account if this is set to true. If the option is absent or if it's set to false, it's just business as usual.

### CODE REVIEW

**Comment by rafaelks** [comment]
@sowenjub We appreciate your PR, but this is something already implemented in our servers as a setting `Accounts_RegistrationForm`. Can you please, update your implementation to follow the `Accounts_RegistrationForm` value? The options are: `Public`, `Disabled` and `Secret URL`. It's OK if you only want to implement `Public` and `Disabled` for now.

**Comment by sowenjub** [comment]
@rafaelks 👍

**Comment by sowenjub** [comment]
@rafaelks Updated to use the Accounts_RegistrationForm. I assumed people using a secret url wouldn't want users who simply downloaded the app to be able to register.

**Comment by rafaelks** [comment]
Looking good @sowenjub! Just need to resolve the conflicts now, I suppose. Thank you!

**Comment by sowenjub** [comment]
@rafaelks [https://github.com/RocketChat/Rocket.Chat.iOS/pull/850/commits/a5646ba7bae1d433c0c9bd431bb8a34a3abdbc1d](this merge) should fix it, but it stills displays the same conflicts, I'm a bit puzzled as to what to do next

File: Rocket.Chat/Controllers/Auth/ConnectServerViewController.swift [hunk 1]
```diff
@@ -65,6 +67,7 @@ final class ConnectServerViewController: BaseViewController {
         DatabaseManager.cleanInvalidDatabases()
 
         if let applicationServerURL = AppManager.applicationServerURL {
+            labelSSLRequired.isHidden = true
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Auth/ConnectServerViewController.swift]: This should not be in this PR, right?

File: Rocket.Chat/Managers/AppManager.swift [hunk 2]
```diff
@@ -35,5 +35,4 @@ struct AppManager {
     static var supportsMultiServer: Bool {
         return applicationServerURL == nil
     }
-
```
Comment (rafaelks) [inline Rocket.Chat/Managers/AppManager.swift]: 🤔

File: Rocket.Chat/Models/AuthSettings.swift [hunk 3]
```diff
@@ -30,6 +30,17 @@ final class AuthSettings: BaseModel {
     @objc dynamic var isGoogleAuthenticationEnabled = false
     @objc dynamic var isLDAPAuthenticationEnabled = false
 
+    // Registration
+    enum RegistrationFormAccess: String {
+        case isPublic = "Public", isDisabled = "Disabled", isSecretURL = "Secret URL"
```
Comment (rafaelks) [inline Rocket.Chat/Models/AuthSettings.swift]: 1. Break each case per line, please
2. Move the enum to outside of the class, please (top)

File: Rocket.Chat/Models/AuthSettings.swift [hunk 4]
```diff
@@ -30,6 +30,17 @@ final class AuthSettings: BaseModel {
...
+    }
+    @objc dynamic var rawRegistrationForm: String?
+    var registrationForm: RegistrationFormAccess {
+        guard let rawValue = rawRegistrationForm,
+            let value = RegistrationFormAccess(rawValue: rawValue) else { return .isPublic }
```
Comment (rafaelks) [inline Rocket.Chat/Models/AuthSettings.swift]: Could you change for this?

```swift
guard let rawValue = rawRegistrationForm,
      let value = RegistrationFormAccess(rawValue: rawValue)
else {
     return .isPublic
}

File: Rocket.Chat/Controllers/Auth/AuthViewController.swift [hunk 5]
```diff
@@ -47,6 +47,8 @@ final class AuthViewController: BaseViewController {
 
     @IBOutlet weak var activityIndicator: UIActivityIndicatorView!
 
+    @IBOutlet var accountCreationFields: [UIView]!
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Auth/AuthViewController.swift]: What's `accountCreationFields`? Shouldn't it be just the `Register button`?

Comment (sowenjub) [reply Rocket.Chat/Controllers/Auth/AuthViewController.swift]: Changed it to just the button

File: Rocket.Chat/Controllers/Auth/ConnectServerViewController.swift (hunk 1 above)
Comment (sowenjub) [reply Rocket.Chat/Controllers/Auth/ConnectServerViewController.swift]: Right

File: Rocket.Chat/Managers/AppManager.swift (hunk 2 above)
Comment (sowenjub) [reply Rocket.Chat/Managers/AppManager.swift]: A space disappeared when I switched from using a custom Info.plist to server values. I put it back 😛

File: Rocket.Chat/Models/AuthSettings.swift (hunk 3 above)
Comment (sowenjub) [reply Rocket.Chat/Models/AuthSettings.swift]: Alright.
Not sure about making that enum global but that's your call.

File: Rocket.Chat/Models/AuthSettings.swift (hunk 4 above)
Comment (sowenjub) [reply Rocket.Chat/Models/AuthSettings.swift]: Ok for aligning vertically but are you sure about the else on a new line? That's the only change I didn't make 😈

Comment (rafaelks) [reply Rocket.Chat/Models/AuthSettings.swift]: @sowenjub Yes, please! 😂 That's how we're doing in other places of the app for guards with multiple lines: https://github.com/RocketChat/Rocket.Chat.iOS/blob/1aba0fcdf2993e108925ca8b39baec79fa829bf0/Rocket.Chat/OAuthManager.swift#L50-L61.

Comment (sowenjub) [reply Rocket.Chat/Models/AuthSettings.swift]: Alright, alright, pushed

**Comment by rafaelks** [comment]
Thanks!!!

**Comment by rafaelks** [comment]
This is looking good now @sowenjub, just added a few tests, will wait for CI and merge it, thanks! 👍

**Comment by sowenjub** [comment]
Awesome, thanks for adding the tests @rafaelks

**Pull request merged**